go/worker/compute/executor/committee/node.go

```diff
@@ -771,7 +787,7 @@ func (n *Node) handleScheduleBatch(force bool) {
 			return roundCtx, nil, nil, nil, errNotTxnScheduler
 		}
 
-		_, roundResults, err := n.getRtStateAndRoundResults(roundCtx, n.commonNode.CurrentBlockHeight)
+		_, roundResults, _, err := n.getRuntimeRoundData(roundCtx, n.commonNode.CurrentBlockHeight, n.commonNode.CurrentBlock.Header.Round)
```

Comment:
Should also make sure to propose an empty block if there are any events and no transactions.